### DISCUSSION TITLE
Videos from Loom not downloading

### DISCUSSION THREAD

**Discussion by angeliti999** [aclap-dev/video-downloadhelper#1434] [Help and Q&A]
### Your question

I'm trying to download a Loom video using the Video DownloadHelper extension, and it fails.

The extension is able to detect the video:

<img width="289" alt="image" src="https://github.com/user-attachments/assets/17e29e02-b47d-4b48-999a-617ad9e8e55b">

However, when I click try download any of the RawHls varaiants, it fails:

<img width="283" alt="image" src="https://github.com/user-attachments/assets/33430fb4-f43c-4d16-80e7-d71a9fbf559d">

And when I try to download the HTTPMedia one, it doesn't fail, but the downloaded file is only 404Kb (about 4 seconds of video).

Any ideas? Could this be the same DRM issue discussed in this other thread?

https://github.com/aclap-dev/video-downloadhelper/discussions/1416

### Addon & CoApp version and details

version: 9.1.1.0
target: microsoft
channel: stable
lang: es-ES
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"34c926b1c9934431"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36 Edg/130.0.0.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 7,
    "successfull_dl": 30
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

You're going to have to give us the URL of the video in question.

**Reply by GrampaWildWilly** [reply]
By the way, that other thread you mentioned does not discuss DRM.  Another user raised the possibility of DRM being involved, but it's highly unlikely that's true.

**Reply by angeliti999** [reply]
The video is embedded inside a Skool group, so it's not public. But I've recorded a video showing the issue, and showing how Vimeo videos from the same group work fine (only Loom videos cannot be downloaded):

https://share.vidyard.com/watch/FtVNS56vQRUZURrhos7PVZ?

I can also provide any other info you need.

**Reply by GrampaWildWilly** [reply]
OK.  It's too bad I can't get into Skool myself, although VDH downloaded your video no problem.  Here's what I would do if I could get into the site.

1. Copy the URL of your Skool page into your clipboard.
2. Open a new, empty browser window.
3. Open the Developer Tools.  On Firefox, this is either F12 or Ctrl+Shift+i or menu navigation
    Tools -> Browser Tools -> Web Developer Tools.
4. On Firefox, I have an option to undock the tools window from the web page, so the web page occupies its own browser tab & the Developer Tools are in their own separate window.  Here's a screenshot of what it looks like on my system.  This is showing the Developer Tools on my second monitor:

![#01](https://github.com/user-attachments/assets/1396bba9-4d00-4728-a145-82cb8cfaf1ff)

Be sure you select the `Network` option.  That sets the Developer Tools to display the Network Monitor.

That image is surely too small to read.  Just click it.  GitHub will open the image in its own browser window.  You will be able to use the browser's own magnifier (in Firefox that's Ctrl-+) to make it legible.
5. In your regular browser window, paste your URL into the URL bar & hit Enter.  The page will load.  Simultaneously, the Network Monitor will load in the other window.
6. In the Network Monitor, execute the menu entry I'm showing in the following image:

![#02](https://github.com/user-attachments/assets/82c725e9-ce16-4587-abb8-3027a8a09b60)

That will launch a standard operating system file selection dialog that allows you to save the HAR file somewhere on your system.  (HAR stands for HTML archive.  It's just a plain text file that you can read in any text editor.)
7. That is probably a large file.  Zip it.
8. Post the zip archive here.

With any luck, that will contain some information that I can use to help you further.  I will use another option on that menu where you saved the HAR.  I will use `Import HAR File` to open your Network Monitor on my system.  I'm hoping I will be able to give you advice on what to do next.  It's too bad I can't get into the Skool site & just show you what to try, but we'll just have to live with this.

**Reply by angeliti999** [reply]
Would it be better if I sent you my Skool credentials and the URL of the problematic video? That way you could log in and debug the issue using with all the tools you currently use.

I don't have anything special in Skool, so I don't mind. Just give me an email or another way of contact and I will be happy to send everything your way.

Thanks for all your help!

**Reply by GrampaWildWilly** [reply]
No.  That is against corporate policy for VDH.  They have posted a number of times over the years that sharing credentials is against their policy.  Even though I don't work for them, I take my cue from them.  I won't do it either.  Let's see if sharing your Network Monitor turns up anything useful.

**Reply by angeliti999** [reply]
I didn't know about that policy, sorry.

Here is the requested HAR file:

https://1drv.ms/u/s!AnlX3jbeyzlonrJ0op_9xJJ6WsnpXQ?e=vQ8Eqo

Let me know if you need anything else from my side.

**Reply by GrampaWildWilly** [reply]
This is progress.  In that Network Monitor, there is an HLS master manifest.  It is the entry I've highlighted in this image:

![#01](https://github.com/user-attachments/assets/ebb652c1-aeb0-4dce-bcf8-ab016f5045a0)

You can get the URL of that object by popping up the context menu on it, as I'm showing there, & executing the `Copy URL` entry on the cascaded submenu.  I took that URL & ran it through ffprobe, giving these rather useless results:

[ffprobe.txt](https://github.com/user-attachments/files/17639716/ffprobe.txt)

You can see in there that I'm getting 403 Forbidden.  I'm assuming that is because I am not logged on there.  But you would be.  So you should probably get much different results.  Run ffprobe on that manifest.  Of course, that would be after you go through the process of opening the Network Monitor & visiting the web page again, like you did to generate the HAR you posted.  The command looks like this:

ffprobe -strict experimental "URL" 2>"Captured log file"

You would paste the URL from what you retrieved from the Network Monitor.  The `Captured log file` is the name of a file you will choose on your system.  It will be a plain text file so give it a file extension of `.txt`.  Otherwise, you can place it in any directory on any partition in your system, wherever you like.  Run the command & post the captured log file.

The quotation marks are important.  Be sure you surround the URL & the file name with quotation marks.

`-strict experimental` tells ffprobe to recognize captions if there are any.  I believe there are some, but we'll get to that further along in our discussion.

To find ffprobe, look in your VDH settings:

![#02](https://github.com/user-attachments/assets/4dda05a9-88c6-4ccf-9c1d-0d3659a1cc2d)

Your ffprobe will be in the directory the VDH settings show on your system.

I am very optimistic we will succeed in downloading this content.

**Reply by angeliti999** [reply]
I'm also getting error 403 because I'm running ffprobe from the CMD, and I guess it should be ran from inside the browser, which is who currently has the permissions to access the file.

Is it possible to do that?

**Reply by GrampaWildWilly** [reply]
Ah.  Well, I think this can be fixed like this.  When you paste the URL of the manifest into your ffprobe command, insert this:

`https://userid:password@URL`

Now there's a trick here.  As you can see, the @ sign separates your id/pw from the URL.  But what if your id or pw contains @?  This could happen if your id is your E-mail address.  Every E-mail address contains @.  So replace every @ in your id & pw with `%40`.

Let's see if that works.

You can't execute a command like ffprobe from inside the browser.  The command prompt & the browser are just 2 different platforms.  They're separate.  Let's see if the id/pw trick makes it work.

**Reply by GrampaWildWilly** [reply]
Oh.  It's possible your user ID & password could appear in your ffprobe output.  Be sure you edit that out of the file before you post it.  The output file is just plain text so you can open it in your text editor & replace those strings with xxx.

**Comment by Mattmjs** [comment]
The dev has said there will be a fix but no time when this will happen. If you're not in a hurry then
I'd sit and wait for it. But if you have limited time before losing access on Loom another tool in yt-dlp works.

**Reply by angeliti999** [reply]
So it's a known bug which will be fixed in the near future? In that case, I guess I can wait. But would be great if you could explain in more detail how to use that "other tool in yt-dlp" which currently works. Thanks!

**Reply by GrampaWildWilly** [reply]
I don't know if anybody else has noticed that @paulrouget Paul, our developer, has not posted here in about 3 months.  I am beginning to suspect he does not work for VDH any more.  I have no inside information on this.  I'm just looking at his total failure to deal with the many issues we've raised here in the past 3 months & I'm getting suspicious.  I sure would like to hear one way or the other what is going on.  This situation is quite unacceptable.

**Reply by Mattmjs** [reply]
It's another tool that you run from the command prompt. For example I found in your HAR file an embed link.

This brings up the formats
 **yt-dlp -F https://www.loom.com/embed/7f7b9de7b703438ab0e641ad11f009e4?autoplay=0&hide_owner=true&hide_share=true&hide_title=true&hideEmbedTopBar=true**

Choose from the formats it finds, you enter format code as you see it.
**yt-dlp -f format code -o "filename.mp4" https://www.loom.com/embed/7f7b9de7b703438ab0e641ad11f009e4?autoplay=0&hide_owner=true&hide_share=true&hide_title=true&hideEmbedTopBar=true**

**Reply by angeliti999** [reply]
This tool worked great, thanks, @Mattmjs! The format that worked for me was http-transcoded.

Cannot this be integrated into Video DowloadHelper?

**Reply by tolgaduran** [reply]
@angeliti999 you can use this tool too => https://www.saveplays.com/loom-downloader/
